definitions/checks/system_registration.rb

```diff
@@ -5,16 +5,26 @@ class Checks::SystemRegistration < ForemanMaintain::Check
     tags :default
 
     confine do
-      file_exists?('/etc/rhsm/rhsm.conf') &&
-        !feature(:foreman_server) &&
-        feature(:foreman_proxy)
+      feature(:downstream) && file_exists?('/etc/rhsm/rhsm.conf')
```

Comment:
This check was for proxy only before and now is satellite server only. Is it okay?

Reply:
IMO feature(:downstream) has confusing behavior and should be fixed (probably not as part of this PR)

Reply:
Yes, previously this check was for proxy. According to BZ details, it should restrict server as well as proxy upgrading in that case and that's why previous condition was removed in initial commit.

My bad, due to my changes, it is now restricting only on server.  My intention behind adding `feature(:downstream)` in config was because there is nothing like self registered term in upstream.

Reply:
And it made perfect sense to use the `fetaure(:downstream)`. And that's why I suggested there is a bug in the feature and it should behave as you expected. `:downstream` shouldn't mean `:downstream_server` and I think that should be fixed. What do you think @iNecas?

@kgaikwad technically I think you can have self-registered Katello, but it does not make sense to talk about support there. Shouldn't the downstream detection stay?

Reply:
@mbacovsky, yes it is needed. Reverting commit back to previous :-)

Reply:
I think `feature(:downsteam)` should be fixed to handle both the server and the capsule. Let's fix that behavior in separete PR, but doing this PR in a way that handles the proxy properly

Reply:
For now, I am removing `feature(:downstream)` from confine as it won't allow to run this check on capsule.

Reply:
I'm tracking the :downstream feature problem in https://projects.theforeman.org/issues/25549.

@kgaikwad, @iNecas  is this check acceptable in upstream? I think we can have Katello registered to itself with subscription-manager. In that case we shouldn't refer to the KBA. If it makes no sense in upstream, we should fix the confine condition or at least add #FIXME and reference the issue.

Reply:
I think in long term we should also have capsule as a package in capsule repositories as we have satellite as package. This will make it very easy to detect that its downstream capsule.

Reply:
@mbacovsky, @upadhyeammit,
Updated condition in confine block. Please verify and let me know if it's serve purpose for now. or any better way to handle it.